### PR TITLE
New CMS Blocks & Styling enhancements

### PR DESCRIPTION
**Pull request by sebazai**
https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/introducing-attributes-and-editable-fields/
https://developer.wordpress.org/block-editor/reference-guides/richtext/

### CODE REVIEW

File: services/cms/src/blocks/BlockWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Wrapper for edible blocks.
```
Comment (sebazai) [inline services/cms/src/blocks/BlockWrapper.tsx:2]: **yummy**

File: shared-module/src/styles/componentStyles.ts [hunk 2]
```diff
@@ -6,7 +6,6 @@ import { defaultContainerWidth } from "./constants"
 // in a page to be inside a container. Some elements need the entire width
 // of the page.
 export const normalWidthCenteredComponentStyles = css`
-  width: 100%;
```
Comment (sebazai) [inline shared-module/src/styles/componentStyles.ts:9]: This causes issues with CMS justify center. i.e.
https://github.com/rage/secret-project-331/pull/294/files#diff-466d275516a10e001cafe8f4b266d13c5fbbddf8ff0fff84d85276357bf044b9R13

Removing this does not affect anything, as max-width > width and on exercises we unset the max-width thus they'll default to full width from parents.

File: services/cms/src/styles/Gutenberg/editor-styles.scss [hunk 3]
```diff
@@ -1,6 +1,4 @@
 .editor-styles-wrapper {
-  font-family: $default-font;
-  font-size: $editor-font-size;
```
Comment (sebazai) [inline services/cms/src/styles/Gutenberg/editor-styles.scss:3]: This probably caused the mismatch on latex snapshot e2e tests.

File: services/cms/src/styles/Gutenberg/style.scss [hunk 4]
```diff
@@ -25,25 +47,10 @@
...
+  // Do we want to support advanced content editing with a mobile phone?
   display: none;
   /* stylelint-disable-next-line */
   @include break-small() {
     display: block;
   }
 }
-
-/**
- * Animations
```
Comment (sebazai) [inline services/cms/src/styles/Gutenberg/style.scss:36]: These are unnecessary as we don't use the edit-post wordpress package.

File: services/cms/src/blocks/HeroSection/HeroSectionEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import React from "react"
+
+import BlockWrapper from "../BlockWrapper"
+
+const ALLOWED_NESTED_BLOCKS = ["core/heading", "core/buttons", "core/paragraph"]
+const HERO_SECTION_TEMPLATE: Template[] = [
+  ["core/heading", { placeholder: "Welcome to...", level: 1, textAlign: "center" }],
+  ["core/paragraph", { placeholder: "Insert short sales speech...", align: "center" }],
+  [
+    "core/buttons",
```
Comment (sebazai) [inline services/cms/src/blocks/HeroSection/HeroSectionEditor.tsx:12]: core/button deprecated, so if we want one button we still should use core/buttons with innerblocks.

Comment (sebazai) [reply services/cms/src/blocks/HeroSection/HeroSectionEditor.tsx:13]: Hardcoded button to use shared-module primary button

File: services/cms/src/blocks/ChapterHeroSection/ChapterHeroSectionEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  ["core/heading", { placeholder: "Chapter heading...", textAlign: "center", level: 2 }],
+  ["core/heading", { placeholder: "Chapter subheading...", textAlign: "center", level: 3 }],
+]
+
+const ChapterHeroSectionEditor: React.FC<BlockEditProps<Record<string, never>>> = ({
+  clientId,
+}) => {
+  return (
+    <BlockWrapper id={clientId}>
+      <InnerBlocks template={CHAPTER_HERO_SECTION_TEMPLATE} allowedBlocks={ALLOWED_NESTED_BLOCKS} />
```
Comment (nygrenh) [inline services/cms/src/blocks/ChapterHeroSection/ChapterHeroSectionEditor.tsx:18]: richtext

File: services/cms/src/blocks/CourseObjectiveSection/CourseObjectiveSectionEditor.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import { InnerBlocks } from "@wordpress/block-editor"
+import { BlockEditProps, Template } from "@wordpress/blocks"
+import React from "react"
+
+import BlockWrapper from "../BlockWrapper"
+
+const ALLOWED_NESTED_BLOCKS = ["core/heading", "core/buttons", "core/button", "core/paragraph"]
+const COURSE_OBJECTIVE_SECTION_TEMPLATE: Template[] = [
+  ["core/heading", { content: "In this course you'll...", level: 2, textAlign: "center" }],
```
Comment (nygrenh) [inline services/cms/src/blocks/CourseObjectiveSection/CourseObjectiveSectionEditor.tsx:9]: RichText

File: services/cms/src/blocks/HeroSection/HeroSectionEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  setAttributes,
+}) => {
+  const { heroHeading } = attributes
+  return (
+    <BlockWrapper id={clientId}>
+      <RichText
+        className="has-text-align-center wp-block-heading"
+        tagName="h1"
+        value={heroHeading}
+        onChange={(value: string) => setAttributes({ heroHeading: value })}
```
Comment (nygrenh) [inline services/cms/src/blocks/HeroSection/HeroSectionEditor.tsx:28]:
```suggestion
        onChange={(value: string) => setAttributes({ title: value })}
```

File: services/cms/src/blocks/HeroSection/index.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  heroHeading: string
+}
+
+const HeroSectionConfiguration: BlockConfiguration<HeroSectionAttributes> = {
+  title: "Hero Section",
+  description:
+    "A hero section is a full screen section typically consisting of a background image, or animations, with text and sometimes a call to action.",
+  category: "design",
+  attributes: {
+    heroHeading: {
```
Comment (nygrenh) [inline services/cms/src/blocks/HeroSection/index.tsx:16]:
```suggestion
    title: {
```

File: services/cms/src/components/editors/GutenbergEditor.tsx [hunk 10]
```diff
@@ -127,38 +121,6 @@ const GutenbergEditor: React.FC<GutenbergEditorProps> = ({
...
-        .wp-block[data-type="moocfi/exercise"],
-        .wp-block[data-type="moocfi/exercise-task"] {
-          /* Exercises are full width */
-          max-width: unset;
-        }
-
-        /* Give space for sidebar so that it won't accidentally overlap the main editor */
-        .editor__content {
-          ${giveSpaceToSidebarStyles}
-        }
```
Comment (nygrenh) [inline services/cms/src/components/editors/GutenbergEditor.tsx:161]: Move these to line 175

File: services/cms/tsconfig.json [hunk 11]
```diff
@@ -15,6 +15,6 @@
     "isolatedModules": true,
     "jsx": "preserve"
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": ["next-env.d.ts", "additional.d.ts", "**/*.ts", "**/*.tsx"],
```
Comment (nygrenh) [inline services/cms/tsconfig.json:18]: Remove

File: services/course-material/tsconfig.json [hunk 12]
```diff
@@ -22,5 +22,5 @@
     "isolatedModules": true
   },
   "exclude": ["node_modules"],
-  "include": ["next-env.d.ts", "katex.d.ts", "**/*.ts", "**/*.tsx"]
+  "include": ["next-env.d.ts", "additional.d.ts", "katex.d.ts", "**/*.ts", "**/*.tsx"]
```
Comment (nygrenh) [inline services/course-material/tsconfig.json:25]: Remove

File: services/example-exercise/tsconfig.json [hunk 13]
```diff
@@ -22,5 +22,5 @@
     "isolatedModules": true
   },
   "exclude": ["node_modules"],
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"]
+  "include": ["next-env.d.ts", "additional.d.ts", "**/*.ts", "**/*.tsx"]
```
Comment (nygrenh) [inline services/example-exercise/tsconfig.json:25]: Remove

File: services/headless-lms/src/bin/seed.rs [hunk 14]
```diff
@@ -730,90 +730,10 @@ async fn seed_cs_course_material(conn: &mut PgConnection, org: Uuid) -> Result<U
...
-                                        ]
-                                    }
-                                ]
-                            }
-                        ]
-                    }
-                ],
-            },
+            GutenbergBlock::empty_block_from_name("moocfi/hero-section".to_string()),
+            GutenbergBlock::empty_block_from_name("moocfi/course-objective-section".to_string()),
```
Comment (nygrenh) [inline services/headless-lms/src/bin/seed.rs:734]: Set the columns inner blocks

File: services/main-frontend/tsconfig.json [hunk 15]
```diff
@@ -15,6 +15,6 @@
     "isolatedModules": true,
     "jsx": "preserve"
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": ["next-env.d.ts", "additional.d.ts", "**/*.ts", "**/*.tsx"],
```
Comment (nygrenh) [inline services/main-frontend/tsconfig.json:18]: remove

**Review by nygrenh — CHANGES_REQUESTED** [review]
LandingPageHeroSection on landing page, HeroSection on other pages

**Review by nygrenh — APPROVED** [review]

**Pull request merged**